### PR TITLE
[travis] [coq] Complete 4.06.0 support.

### PR DESCRIPTION
**Pull request by ejgallego**
Due to an API change in laglgtk, we need to update CoqIDE. We use a
makefile hack so it can compile with lablgtk < 2.8.16, another option
would be to require 2.8.16 as a minimal dependency.

### CODE REVIEW

**Comment by ejgallego** [comment]
Let's wait to merge this until OCaml 4.06.0 is out, as it needs a tweak.

**Comment by ejgallego** [comment]
@Zimmi48 this should be ready to apply on top of 8.7 as to see if Travis is going to like it. Only change remaining is move `4.06.0+trunk` to `4.06.0`, which I guess it makes sense.

**Comment by ejgallego** [comment]
Ok, this is ready to merge, and indeed it _must_ be merged as OCaml people thought that removing the 4.06.0 compiler switch we are using was a good idea ....

File: .travis.yml [hunk 1]
```diff
@@ -30,17 +30,19 @@ env:
   - NJOBS=2
   # system is == 4.02.3
   - COMPILER="system"
+  - COMPILER_BE="4.06.0"
```
Comment (maximedenes) [inline .travis.yml]: Sorry, what does `BE` stand for?

File: Makefile.ide [hunk 2]
```diff
@@ -123,6 +123,15 @@ ide/%.cmx: ide/%.ml
 	$(SHOW)'OCAMLOPT  $<'
 	$(HIDE)$(OCAMLOPT) $(COQIDEFLAGS) $(OPTFLAGS) -c $<
 
+# We need to compile this file without -safe-string due mess with
+# lablgtk API. Other option is to require lablgtk >= 2.8.16
```
Comment (maximedenes) [inline Makefile.ide]: Does Travis test the compilation with both lablgtk versions (i.e. the minimum we require and the latest) ?

File: library/coqlib.mli [hunk 3]
```diff
@@ -208,4 +208,5 @@ val coq_iff_ref : global_reference lazy_t
 
 (* Deprecated functions *)
 val gen_reference :  message -> string list -> string -> global_reference
-[@@ocaml.deprecated "Please use Coqlib.find_reference"]
+(* Cannot mark as such here, mark should go in API.ml      *)
```
Comment (maximedenes) [inline library/coqlib.mli]: Out of curiosity, how is this change related to OCaml 4.06.0?

Comment (ejgallego) [reply library/coqlib.mli]: Better propagation of warning in module aliases.

File: Makefile.ide (hunk 2 above)
Comment (ejgallego) [reply Makefile.ide]: I think it does not, will add some bounds.

File: .travis.yml (hunk 1 above)
Comment (ejgallego) [reply .travis.yml]: Bleeding Edge.

File: Makefile.ide (hunk 2 above)
Comment (ejgallego) [reply Makefile.ide]: The PR seems ready to go, [it build correctly on my Travis], the bounds are in place.

**Comment by SkySkimmer** [comment]
Synch the gitlab yml?

**Comment by ejgallego** [comment]
> Synch the gitlab yml?

Ok, but that will come in a separate PR as not to reTravis this [which is kinda urgent].

I am not looking at Gitlab as we did not setup gitlab/coq yet, but we should at some point.

**Comment by maximedenes** [comment]
This seems to now fail on some deprecation warnings.

**Comment by SkySkimmer** [comment]
Since this needs retravis I insist on gitlab update ;)
I guess newer ocaml is smarter about deprecation in some way, otherwise why would it fail with 4.06 but not with 4.02.3?

**Comment by ejgallego** [comment]
Can we merge #6098 first so we solve deprecation once and forall?

**Comment by ejgallego** [comment]
Ok, should be good now. Hard to understate how much of a disaster the API.mli file is.

**Comment by Zimmi48** [comment]
So this is basically reverting #6098?

**Comment by ejgallego** [comment]
> So this is basically reverting #6098?

Not at all.

**Comment by Zimmi48** [comment]
> Not at all.

Sorry for my earlier dumb comment. (I should refrain from commenting on my phone. I had forgotten what #6098 was really about and remembered what I thought it was about some time before by just reading the title. I don't know if anyone but me can understand this explanation though :P).

**Comment by ejgallego** [comment]
> Sorry for my earlier dumb comment. (I should refrain from commenting on my phone. I had forgotten what #6098 was really about and remembered what I thought it was about some time before by just reading the title. I don't know if anyone but me can understand this explanation though :P).

Oh well, I think it is a very legitimate question. My answer just meant to point out that there is really no relationship between both commits [the problem in 4.06 exists independently of #6098]
